### DISCUSSION TITLE
Is `DrawTrianglesOptions` missing a `GeoM`?

### DISCUSSION THREAD

**Discussion by ajzaff** [hajimehoshi/ebiten#1809] [Q&A]
_This may be filed under the tag of questions but I am interested in the change if feasible._

I would like to draw object meshes using the `DrawTriangles*` API. Without a GeoM however, I must compute `vertices+geoM` every update. Have I misused the API or is this working as intended? Is there some technical reason why we cannot pass a `GeoM` for these Draw calls.

Thanks!

**Comment by hajimehoshi** [comment]
Right, `DrawTrianglesOptions` doesn't have `GeoM` and you have to adjust vertices by yourself. Actually with `GeoM`, the calculation happens every update inside.

> Is there some technical reason why we cannot pass a GeoM for these Draw calls.

No technically reason. 1) It's just because `DrawTriangles` users should have more controls of vertices than `DrawImage`, then I thought we don't have to have it. 2) Another reason is performance. If `GeoM` exists in the option, performance is a little degraded due to some overhead.
